.github/workflows/tests.yml

```
@@ -0,0 +1,42 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Tests
+
+on:
+  push:
+    branches:
+    # Check all branches
+  pull_request:
+    # Check all PR
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

I guess this line simply installs all packages in requirements.txt. However, because requirements.txt does not contain the MDTools package itself, it will not install MDTools. I'm not sure, if flake8 actually requires the target package (i.e. the package to be tested) to be installed. But pytest most likely requires this (I don't know how pytest will otherwise run the tests).

So, I guess we also have to install the MDTools package itself via
```suggestion
        python -m pip install .
```

---

This all _should_ work without installing MDTools, but I will create some very basic test jut to make sure.

---

It is not necessary to install the package. Tests run successfully: https://github.com/ezavod/mdtools/runs/2940951065

---

Ah nice, I see. And there are also all the warnings about blank lines and other formatting stuff that I were expecting.

Do we then need to install the dependencies of MDTools contained in the requirement file at all? I mean, if we do not need to install MDTools, it might be that we have not to install anything (except for flake8 and pytest)

---

We need to have the dependencies installed. Actions always starts with a clean setup and the dependencies are needed for the tests.